operate/services/manage.py

```diff
@@ -1327,10 +1327,37 @@ def terminate_service_on_chain_from_safe(  # pylint: disable=too-many-locals
 
         if counter_current_safe_owners == counter_instances:
             self.logger.info(
-                "Service funded for safe swap"
+                "[SERVICE MANAGER] Funding agent EOA for Safe swap."
             )  # TODO: is this safe swapping needed anymore?
-            # TODO the funding below seems too much just for safe swapping!
-            self.funding_manager.fund_service_initial(service)
+
+            requirements = ChainAmounts(
+                {
+                    chain: {
+                        current_safe_owners[0]: {
+                            ZERO_ADDRESS: chain_data.user_params.fund_requirements[
+                                ZERO_ADDRESS
+                            ].agent
+                        }
+                    }
+                }
+            )
+            balances = ChainAmounts(
+                {
+                    chain: {
+                        current_safe_owners[0]: {
+                            ZERO_ADDRESS: get_asset_balance(
+                                ledger_api=sftxb.ledger_api,
+                                asset_address=ZERO_ADDRESS,
+                                address=service.agent_addresses[0],
```

Comment:
`current_safe_owners[0]` and `service.agent_addresses[0]` will always be the same right?

Reply:
Yes, this block of code enters when `counter_current_safe_owners == counter_instances`. Assuming the config.json hasn't been manipulated.